internal/elasticsearch/status.go

```diff
@@ -430,7 +434,7 @@ func (er *ElasticsearchRequest) pruneMissingNodes(status *api.ElasticsearchStatu
 			"node-name":    nodeName,
 		}
 
-		nodePodList, err := GetPodList(cluster.GetNamespace(), matchingLabels, er.client)
+		nodePodList, err := pod.List(context.TODO(), er.client, cluster.GetNamespace(), matchingLabels)
```

Comment:
we shouldn't need to provide the context to `pod.List` we should have an informed decision that certain operators are going to use certain contexts (also I'm 99% sure we only use `context.TODO()` anyways)

Reply:
The entirety of this PR is using `context.TODO()` as it was before as per we are not using context properly down the stack from top (controller) to bottom (k8s calls). I try to keep the door open for a later refactoring, where  we use context properly. `context.TODO()` is basically the marker to fix this later.

Reply:
we'll still be using specific contexts based on the operation we're doing. no need to pass it down, we should be making an informed decision at the actual client call level

Reply:
We haven't used `context.Context` effectively in entire EO so far. Everytime we had to pass one, we used/use `context.TODO`. I am not sure why we don't have a single reconciliation loop context to share common stuff, but this is a question for another refactoring if in need for such. The only purpose to pass `context` here to our `internal/manifests` function is to leave the door open in case we want to use `context` later. The k8s API makes it easy to use context for doing things like request cancelation. Since `internal/manifests` represents a set of wrappers for the k8s API that consist, it is only consistent to passthrough context from the caller site.